starfish/core/types/_spot_finding_resutls.py

```diff
@@ -0,0 +1,86 @@
+from typing import List, Mapping, Optional, Tuple
+
+from starfish.core.types import Axes, SpotAttributes
+
+AXES_ORDER = (Axes.ROUND, Axes.CH)
+
+
+class SpotFindingResults:
+    """
+    Wrapper class that describes the results from a spot finding method. The
+    results dict is a collection of (round,ch) indices and their corresponding measured
+    SpotAttributes.
+    """
+
+    def __init__(self, spot_attributes_list: Optional[List[Tuple]] = None):
+        """
+        Construct a SpotFindingResults instance
+
+        Parameters
+        -----------
+        spot_attributes_list : Optional[List[Tuple[indices, SpotAttributes]]]
+            If spots were fond using Imagestack.transform() the result is a list of
+            tuples (indices, SpotAttributes). Instantiating SpotFindingResults with
+            this list will convert the information to a dictionary.
+        """
+        self._results: Mapping[Tuple, SpotAttributes] = dict()
+        if spot_attributes_list:
+            for indices, spots in spot_attributes_list:
+                self._results[indices] = spots
+
+    def set_spots_for_round_ch(self, indices: Mapping[Axes, int], spots: SpotAttributes
+                               ) -> None:
+        """
+        Add the r,ch indices and corresponding SpotAttributes to the results dict.
+
+        Parameters
+        ----------
+        indices: Mapping[Axes, int]
+            Mapping of Axes to int values
+        spots: SpotAttributes
+            Describes spots found on this tile.
+        """
+        round_ch_index = tuple(indices[i] for i in AXES_ORDER)
+        self._results[round_ch_index] = spots  # type: ignore
+
+    def get_spots_for_round_ch(self, indices: Mapping[Axes, int]) -> SpotAttributes:
+        """
+        Returns the spots found in a given round and ch.
+
+        Parameters
+        ----------
+        indices: Mapping[Axes, int]
+            Mapping of Axes to int values
+
+        Returns
+        --------
+        SpotAttributes
+        """
+        round_ch_index = tuple(indices[i] for i in AXES_ORDER)
+        return self._results[round_ch_index]
+
+    def round_ch_indices(self):
+        """
+        Return all round, ch index pairs.
+        """
+        return self._results.keys()
+
+    def all_spots(self):
+        """
+        Return all SpotAttributes across rounds and chs.
+        """
+        return self._results.values()
+
+    @property
+    def round_labels(self):
+        """
+        Return the set of Round labels in the SpotFindingResults
+        """
+        return list(set(sorted(r for (r, ch) in self.round_ch_indices())))
```

Comment:
if you take a set of sorted, then you lose the sorting.  did you want sorted(set(...))?